### PR TITLE
feat(route): accept timeout to fetch

### LINKED ISSUE #22474: [Feature] Allow timeout to be passed to Route.fetch
Currently it is not possible to specify `timeout` when calling `Route.fetch`. It appears to use the default timeout.

https://playwright.dev/docs/api/class-route#route-fetch

### PR DESCRIPTION
**Pull request by elliots**
Update: test added and docs updated with suggestions.

fixes: #22474

### CODE REVIEW

**Comment by elliots** [comment]
@microsoft-github-policy-service agree [company="UPowr Pty Ltd"]

**Comment by elliots** [comment]
@microsoft-github-policy-service agree company="UPowr Pty Ltd"

File: docs/src/api/class-route.md [hunk 1]
```diff
@@ -503,6 +503,13 @@ If set changes the request URL. New URL must have same protocol as original one.
 Maximum number of request redirects that will be followed automatically. An error will be thrown if the number is exceeded.
 Defaults to `20`. Pass `0` to not follow redirects.
 
+### option: Route.fetch.timeout
+* since: v1.33
+- `timeout` <[float]>
+
+Maximum wait time in milliseconds.
```
Comment (dgozman) [inline docs/src/api/class-route.md:510]:
```suggestion
Request timeout in milliseconds. Defaults to `30000` (30 seconds). Pass `0` to disable timeout.
```

**Review by dgozman — COMMENTED** [review]
This looks good overall! I added a couple of suggestions for the docs.

For the test, we would need something like this: https://github.com/microsoft/playwright/blob/main/tests/page/page-request-intercept.spec.ts#L197-L207, but testing timeout like in this test: https://github.com/microsoft/playwright/blob/main/tests/library/browsercontext-fetch.spec.ts#L711-L721.

Let me know if you have more questions.

File: tests/page/page-request-intercept.spec.ts [hunk 2]
```diff
@@ -206,6 +206,24 @@ it('should fulfill intercepted response using alias', async ({ page, server, isE
...
+    res.writeHead(200, {
+      'content-length': 4096,
+      'content-type': 'text/html',
+    });
+  });
+  await page.route('**/*', async route => {
+    const error = await route.fetch({ timeout: 10 }).catch(e => e);
+    expect(error.message).toContain(`Request timed out after 10ms`);
+  });
+  const error = await page.goto(server.PREFIX + '/slow', { timeout: 20 }).catch(e => e);
```
Comment (elliots) [inline tests/page/page-request-intercept.spec.ts:223]: I added a small (but larger than the inner) timeout here so it wouldn't slow down the tests waiting for it to time out, but if there's a better way?

Comment (dgozman) [reply tests/page/page-request-intercept.spec.ts:223]: This is good. I'd only ask to bump 10 to 1000 and 20 to 2000. Given our experience on GHA bots, 10/20 is waaay too small for non-flaky tests.

**Review by dgozman — COMMENTED** [review]
Just a single comment, and this is good to go. Please also remove the Draft status 😄

File: docs/src/api/class-route.md [hunk 3]
```diff
@@ -503,6 +503,12 @@ If set changes the request URL. New URL must have same protocol as original one.
 Maximum number of request redirects that will be followed automatically. An error will be thrown if the number is exceeded.
 Defaults to `20`. Pass `0` to not follow redirects.
 
+### option: Route.fetch.timeout
+* since: v1.33
```
Comment (elliots) [inline docs/src/api/class-route.md:507]: I see the issue was tagged `1.34`, I'll update this?

Comment (dgozman) [reply docs/src/api/class-route.md:507]: No, it is right on time for 1.33 for now. If anything, I'll bump to 1.34 before the release.

**Review by dgozman — APPROVED** [review]
Thank you for the PR, merging in!

**Pull request merged**